java/core/src/test/java/co/worklytics/psoxy/impl/RESTApiSanitizerImplTest.java

```diff
@@ -590,4 +594,69 @@ public void pseudonymizeWithRegexMatches_nonMatchingRedacted(String input, Strin
         assertEquals(StringUtils.trimToNull(expected),
             transform.map(input, sanitizer.getJsonConfiguration()));
     }
+
+    @CsvSource(value = {
+        "123.234.252.12,c16e:9376:40a8:d00c:9681:6bf0:b8f:19f3",
+        "10.0.0.1,240d:2561:790d:b861:19d:a797:d452:e284",
+        ",",
+        "not an ip," // redact
+    })
+    @ParameterizedTest
+    public void hashIp(String input, String expected) {
+        MapFunction transform = sanitizer.getHashIp(HashIp.builder().build());
+
+        assertEquals(StringUtils.trimToNull(expected),
+            transform.map(input, sanitizer.getJsonConfiguration()));
+    }
+
+    @SneakyThrows
+    @CsvSource(value = {
+        "123.234.252.12,p~wW6TdkCo0AyWgWvwC48Z88ICwFaKIY3roGDJ2lIxURk",
+        "10.0.0.1,p~JA0lYXkNuGEBnaeX1FLihKXznxklfgEYWKWzFUcFLic",
+        ",",
+        "not an ip," // redact
+    })
+    @ParameterizedTest
+    public void encryptIp(String input, String expected) {
+        MapFunction transform = sanitizer.getEncryptIp(EncryptIp.builder().build());
+
+
+        String encrypted = (String) transform.map(input, sanitizer.getJsonConfiguration());
+
+
+        assertEquals(StringUtils.trimToNull(expected),
+            encrypted);
+
+
+        if (StringUtils.isNotBlank(expected)) {
+            MapFunction hashTransform = sanitizer.getHashIp(HashIp.builder().build());
+            String formattedHash = (String) hashTransform.map(input, sanitizer.getJsonConfiguration());
+
+            byte[] hash = Inet6Address.getByName(formattedHash).getAddress();
+
+            Base64.Encoder base64encoder = Base64.getUrlEncoder().withoutPadding();
+            String withoutPrefix = encrypted.substring(2);
+            byte[] decoded = Base64.getUrlDecoder().decode(withoutPrefix.getBytes());
+
+
+            assertEquals(
+                base64encoder.encodeToString(hash),
+                base64encoder.encodeToString(Arrays.copyOfRange(decoded, 0, Md5DeterministicTokenizationStrategy.HASH_LENGTH_BYTES))
+            );
+
+            // TODO: fix the following; UrlSafeTokenPseudonymEncoder can't decode the encrypted value
+            // because it produces a 32-byte hash, but the EncryptIP uses 16-byte hash
+            // options:
+            //  - a different prefix (m~) that indicates 16-byte hash?
```

Comment:
or maybe `e16~` for encrypted w 16-byte IV? 
do we need to encode more info about encryption algorithm here??

or for `encryptIP` we relax the property that the IV == the result of the `hashIP` transform?